server/fleet/integrations.go

```diff
@@ -0,0 +1,427 @@
+package fleet
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/fleetdm/fleet/v4/server/service/externalsvc"
+)
+
+// TeamIntegrations contains the configuration for external services'
+// integrations for a specific team.
+type TeamIntegrations struct {
```

Comment:
For now, Team integrations is a subset of the global integrations. Eventually, they might reach feature parity and the code can be simplified a bit (quite significantly), but even then it might still be better to keep both distinct to be more flexible if we have new features that are only available at one level.